src/System.Net.Http/src/System/Net/Http/Headers/HeaderDescriptor.cs

```diff
@@ -111,7 +114,42 @@ public string GetHeaderValue(ReadOnlySpan<byte> headerValue)
                 }
             }
 
+            if (KnownHeader != null && KnownHeader.Name == KnownHeaders.Location.Name)
+            {
+                // Normally Location should be in ISO-8859-1, ocassionally some servers respond with UTF-8 though
+                if (TryDecodeUtf8(headerValue, out string decoded))
+                {
+                    return decoded;
+                }
+            }
+
             return HttpRuleParser.DefaultHttpEncoding.GetString(headerValue);
         }
+
+        private static bool TryDecodeUtf8(ReadOnlySpan<byte> input, out string decoded)
+        {
+            bool possibleUtf8 = false;
+            for (int i = 0; i < input.Length; i++)
+            {
+                if (input[i] > 127)
+                {
+                    possibleUtf8 = true;
+                    break;
+                }
+            }
```

Comment:
if you care about the optimization, can perform this by comparing integers with some mask instead of going byte by byte. just ensure handling Big/Little Endian

Reply:
My understanding is that Location is used only in case of redirects in which case this does not seem to be a hot path but I'd prefer if someone more familiar with the space commented cc: @davidsh @stephentoub

Reply:
While it'd be nice to vectorize this, I'd rather we do that via an existing helper than duplicating the vectorization code here.  @GrabYourPitchforks created an optimized implementation in ASCIIUtility in corelib.  I'd suggest holding off on vectorizing this here, and instead adding a TODO to utilize those helpers if/when they're made public (it's probably not worth including all of that code into System.Net.Http just for this).